boards/qemu/qemu_riscv32/Kconfig.qemu_riscv32

```
@@ -0,0 +1,7 @@
+# Copyright (c) Antmicro 2024 <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_RISCV32_FE310_XIP
+	bool
+	default y
```

lease, no bool, as the `BOARD_<name>` and `BOARD_<full_identifier>` are constructed here:
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L8
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L13

Some details can be found here: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#common-kconfig-shorthands

Please do:
```suggestion
config BOARD_QEMU_RISCV32
```
as that will ensure any typos or similar wrong selection is caught by CI.